src/components/Table/Table.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from 'react';
+import { Person } from '../../types';
+import { useParams } from 'react-router-dom';
+import { Loader } from '../Loader';
+import { getPeople } from '../../api';
+
+export const Table: React.FC = () => {
+  const { personSlug } = useParams();
+  const [requestState, setRequestState] = useState('loading');
+  const [peopleTable, setPeopleTable] = useState<Person[] | null>(null);
+
+  useEffect(() => {
+    fetch('https://mate-academy.github.io/react_people-table/api/people.json');
+    getPeople()
+      .then((data: Person[]) => {
+        if (data.length === 0) {
+          return [];
+        }
+        const treatedData = data.map(e => {
+          const motherSlug = data.find(el => el.name === e.motherName)?.slug;
+          const fatherSlug = data.find(el => el.name === e.fatherName)?.slug;
+
+          return { ...e, motherSlug: motherSlug, fatherSlug: fatherSlug };
+        });
+
+        return treatedData;
+      })
+      .then(data => {
+        setPeopleTable(data);
+        setRequestState(data.length === 0 ? 'empty success' : 'success');
+      })
+      .catch(() => {
+        setRequestState('error');
+      });
+  }, []);
+
+  return (
+    <div className="block">
+      <div className="box table-container">
+        {requestState.includes('loading') ? (
+          <Loader />
+        ) : (
+          <>
+            {requestState.includes('error') && (
+              <>
+                <p data-cy="peopleLoadingError" className="has-text-danger">
+                  {' '}
+                  Something went wrong
+                </p>
+                {requestState.includes('empty') && (
```

Comment:
I which case will this line run?

Reply:
I think that is about tests applied. But I'll check this again.